### DISCUSSION TITLE
CLI: How to correlate links in a provided urls file to output files?

### DISCUSSION THREAD

**Discussion by rho-cassiopeiae** [gildas-lormeau/SingleFile#1084] [Q&A] [answered]
I use single-file cli, everything works fine, but I can't figure out how to correlate links given in `--urls-file <some_file.txt>` to output files. E.g. there are 4 urls in the file, meaning there are 4 resultant html files. How can I programmatically determine which output file corresponds to which url?

**Comment by gildas-lormeau** [comment, marked as answer]
You could use the `--filename-template` switch and set it to `{url-href-flat}` for example. Alternatively, the URL is stored in the saved page at line 3, e.g. ` url: https://example.com/ `.
